GroceriesApp/GroceryappDesign.sql

```diff
@@ -0,0 +1,158 @@
+CREATE DATABASE GroceriesAppdb;
+Use GroceriesAppdb;
+
+--creating table
+
+DROP TABLE UserDetails
+CREATE TABLE  UserDetails(
+    RegisterId INT IDENTITY(1,1),
+    FirstName VARCHAR(20) NOT NULL,
+    LastName VARCHAR(20) NOT NULL,
+    Email VARCHAR(40) , -- Unique email
+    [Password] VARCHAR(20) NOT NULL,
+    ConfirmPassword VARCHAR(20) NOT NULL,
+    PhoneNumber VARCHAR(15) NOT NULL, -- Unique phone number
+    [Address] VARCHAR(60) NOT NULL, 
+    City VARCHAR(30) NOT NULL,
+    [State] VARCHAR(30) NOT NULL,
+    PinCode VARCHAR(10) NOT NULL,
+	UserType VARCHAR(40) NOT NULL,
+    DateRegistered DATETIME DEFAULT GETDATE(), -- Default to current date and time
+    CONSTRAINT PK_UserDetails_RegisterId PRIMARY KEY (RegisterId),
+    CONSTRAINT UQ_UserDetails_PhoneNumber UNIQUE (PhoneNumber),
+    CONSTRAINT UQ_UserDetails_RegisterEmail UNIQUE (Email)
+);
+
+
+
+INSERT INTO UserDetails(
+    FirstName, LastName, Email, [Password], ConfirmPassword, PhoneNumber,[Address], City, [State], PinCode, UserType
+) VALUES
+    ('John', 'Doe', 'john.doe@example.com', 'password123', 'password123', '+1234567890', '123 Main St', 'New York', 'NY', '10001', 'Customer'),
+    ('Jane', 'Smith', 'jane.smith@example.com', 'password123', 'password123', '+0987654321', '456 Maple Ave', 'Los Angeles', 'CA', '90001', 'Customer'),
+    ('Supplier One', 'Supplier', 'supplier1@example.com', 'password123', 'password123', '+9876543210', '789 Oak St', 'Chicago', 'IL', '60601', 'Supplier'),
+    ('Supplier Two', 'Supplier', 'supplier2@example.com', 'password123', 'password123', '+0123456789', '321 Pine Ave', 'Miami', 'FL', '33101', 'Supplier');
+	
+-- Verify the data insertion
+SELECT * FROM UserDetails;
+
+
+CREATE TABLE Products_Supplier (
+    ProductSupplierId INT IDENTITY(1,1),
+    ProductName VARCHAR(20) NOT NULL,
+    ProductPrice MONEY NOT NULL,
+    ProductDescription VARCHAR(50) NOT NULL,
+    ProductQuantity INT NOT NULL,
+   CONSTRAINT PK_Products_Supplier_ProductSupplierId PRIMARY KEY(ProductSupplierId),	
+);
+
+
+-- Insert dummy data into Products_Supplier
+INSERT INTO Products_Supplier ( ProductName, ProductPrice, ProductDescription, ProductQuantity)
+VALUES 
+( 'Apple', 0.50, 'Fresh Red Apple', 100),
+( 'Banana', 0.30, 'Fresh Yellow Banana', 150),
+( 'Carrot', 0.20, 'Organic Carrot', 200),
+( 'Tomato', 0.40, 'Fresh Red Tomato', 180),
+( 'Cheese', 2.00, 'Cheddar Cheese', 30);
+
+
+
+-- Verify the data insertion
+SELECT * FROM Products_Supplier;
+
+
+
+	CREATE TABLE Categories(
+    CategoryId INT  IDENTITY(1,1),
+	CategoryName VARCHAR(20)  NOT NULL,
+	CONSTRAINT UQ_Categories_CategoryName UNIQUE (CategoryName),
+	CONSTRAINT PK_Categories_CategoryId  PRIMARY KEY(CategoryId),
+
+);
+
+
+INSERT INTO Categories( CategoryName) VALUES
+    ( 'Fruits'),
+    ( 'Vegetables'),
+    ( 'Dairy');
+
+	
+-- Verify the data insertion
+SELECT * FROM Categories;
+
+CREATE TABLE Products(
+ ProductsId INT IDENTITY(1,1),
```

Comment:
What is the difference between Products table and Products_Supplier table?

Both have the same information inserted.
For Example: Apple fruit is inserted as the first record in both the tables. Then what is the use of creating two tables?